include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -1514,11 +1514,13 @@ __pattern_partial_sort_copy(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&&
 
         auto __buf_mid = __buf_first + __out_size;
 
-        __par_backend_hetero::__parallel_partial_sort(
+        auto __future_obj = __par_backend_hetero::__parallel_partial_sort(
             _BackendTag{}, __par_backend_hetero::make_wrapped_policy<__partial_sort_2>(__exec),
             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::read_write>(__buf_first),
             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::read_write>(__buf_mid),
             __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::read_write>(__buf_last), __comp);
+        if constexpr (!decltype(__buf).IsSync{})
```

Comment:
One more way is to add `.wait()` call here without any conditions and de-facto existing synchronizations inside `__parallel_partial_sort` (`__parallel_partial_sort_impl`) due `sycl::accessor` syncrhonizations.